packages/api/lambdas/sf-scheduler.js

```diff
@@ -7,60 +7,62 @@ const { getS3Object, parseS3Uri } = require('@cumulus/common/aws');
 const { SQS } = require('@cumulus/ingest/aws');
 const { Provider, Collection } = require('../models');
 
-/**
- * Builds a cumulus-compatible message and adds it to the startSF queue
- * startSF queue will then start a stepfunction for the given message
- *
- * @param   {Object} event   - lambda input message
- * @param   {Object} context - lambda context
- * @param   {function} cb    - lambda callback
- */
-function schedule(event, context, cb) {
-  const template = get(event, 'template');
+const buildCumulusMeta = (queueName) => ({
+  execution_name: uuidv4(),
+  queueName
+});
+
+function getProvider(provider) {
+  if (provider) {
+    const p = new Provider();
+    return p.get({ id: provider });
+  }
+  return {};
+}
+
+function getCollection(collection) {
+  if (collection) {
+    const c = new Collection();
+    return c.get({ name: collection.name, version: collection.version });
+  }
+  return {};
+}
+
+function buildMessage(event, baseMessage) {
   const provider = get(event, 'provider', null);
   const meta = get(event, 'meta', {});
   const cumulusMeta = get(event, 'cumulus_meta', {});
   const collection = get(event, 'collection', null);
   const payload = get(event, 'payload', {});
-  let message;
+  const queueName = get(event, 'queueName', 'startSF');
+
+  return {
+    ...baseMessage,
+    provider: getProvider(provider),
+    collection: getCollection(collection),
+    meta: merge(baseMessage.meta, meta),
+    payload,
+    cumulus_meta: merge(buildCumulusMeta(queueName), cumulusMeta)
```

Comment:
Would we want `buildCumulusMeta(queueName)` to be the second argument here? As written, if `cumulusMeta.queueName = undefined/null`, then that would overwrite the queueName from ``buildCumulusMeta(queueName)`, correct?

Reply:
You are correct. I don't think we should expect `queueName` to be `null`/empty at this point (because of the changes made to the `buildPayload` function in `rules`, but I'll make that change.

Reply:
https://github.com/nasa/cumulus/pull/1006/commits/fedd15458391c5e5ec69817d96fbdd09b9b2b5da